### PR TITLE
Change OpenURL to use a URL and return error

### PR DESCRIPTION
**Pull request by jadr2ddude**
WIP - changes OpenURL to accept a URL and return error.

### CODE REVIEW

File: widget/hyperlink_test.go [hunk 1]
```diff
@@ -35,19 +41,22 @@ func TestHyperlink_SetText(t *testing.T) {
...
 
 	// test setting functions
-	sUrl = "https://fyne.io"
+	sUrl, err = url.Parse("https://fyne.io")
+	assert.Nil(t, err)
 	hyperlink.SetUrl(sUrl)
 	assert.Equal(t, sUrl, hyperlink.Url)
-	sUrl = "duck.com"
-	uUrl, err := url.Parse(sUrl)
+	sUrl, err = url.Parse("duck.com")
```
Comment (paulhovey) [inline widget/hyperlink_test.go]: Seems like this through line 61 can be taken out, redundant of above test

Comment (jadr2ddude) [reply widget/hyperlink_test.go]: should be fixed by next commit

File: app/app_test.go [hunk 2]
```diff
@@ -29,6 +28,8 @@ func TestCurrentApp(t *testing.T) {
 //	assert.Equal(t, "light", fyne.GlobalSettings().Theme())
 //}
 
+// TODO: fix app to avoid summoning the kraken
```
Comment (jadr2ddude) [inline app/app_test.go]: This test causes strange GLFW errors for me. I believe that there might be a problem with concurrent starting and stopping. Also, this test used to leave the app running forever.

Comment (jadr2ddude) [reply app/app_test.go]: One thought is that we could replace quit with a context passed to Run that would shut the UI down on cancellation.

Comment (jadr2ddude) [reply app/app_test.go]: Should this be moved into a separate issue?

**Comment by jadr2ddude** [comment]
So. . . somehow i ended up with the merge squashing in the process, and squashing in your commits. @andydotxyz is this ok?

File: app/app_windows.go [hunk 3]
```diff
@@ -2,8 +2,14 @@
...
-import "os/exec"
+import (
+	"net/url"
+	"os"
+	"os/exec"
+)
 
-func (app *fyneApp) OpenURL(url string) {
-	exec.Command("rundll32", "url.dll,FileProtocolHandler", url).Run()
+func (app *fyneApp) OpenURL(url *url.URL) {
```
[the remark this answers is not in the document]
Comment (jadr2ddude) [reply app/app_windows.go]: Thanks. Missed this one.

File: cmd/fyne_demo/main.go [hunk 4]
```diff
@@ -51,7 +52,11 @@ func confirmCallback(response bool) {
 
 func showAppDialog(w fyne.Window) {
 	label := widget.NewLabel("Example applications have moved to")
-	link := widget.NewHyperlink("github.com/fyne-io/examples", "https://github.com/fyne-io/examples")
+	u, err := url.Parse("https://github.com/fyne-io/examples")
```
[the remark this answers is not in the document]
Comment (jadr2ddude) [reply cmd/fyne_demo/main.go]: Yes, we really should.

File: widget/hyperlink.go [hunk 5]
```diff
@@ -41,14 +41,14 @@ func (hl *Hyperlink) SetText(text string) {
...
+func (hl *Hyperlink) SetUrl(url *url.URL) {
 	hl.Url = url
 }
 
-// SetUrl sets the URL of the hyperlink, taking in a URL type
-func (hl *Hyperlink) SetUrlFromUrl(url *url.URL) {
-	hl.Url = url.String()
+// SetUrl sets the URL of the hyperlink, taking in a string type
+func (hl *Hyperlink) SetUrlFromString(u string) {
+	hl.Url, _ = url.Parse(u)
```
[the remark this answers is not in the document]
Comment (jadr2ddude) [reply widget/hyperlink.go]: Yeah, will fix.

File: app/app_xdg.go [hunk 6]
```diff
@@ -0,0 +1,17 @@
+// +build !ci
```
[the remark this answers is not in the document]
Comment (jadr2ddude) [reply app/app_xdg.go]: While I do agree that the name could be clearer, neither unix nor freedesktop are fully accurate. There are go-supported *nix systems which fyne does not yet support where xdg-open does not exist. In addition, some freedesktop-based systems prefer some other open command, but maintain support for xdg-open. For now, I think this should be fine. However, changing the name from app to something like url might be more useful.

[the remark this answers is not in the document]
Comment (andydotxyz) [reply app/app_xdg.go]: Good points. We can leave for now and review as it evolves. Such changes won’t affect the API anyhow.

**Comment by jadr2ddude** [comment]
This should be good to merge now.

**Pull request merged**